support/k8s/k8s-workload-registrar/controllers/pod_controller.go

```diff
@@ -0,0 +1,224 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"path"
+
+	"github.com/sirupsen/logrus"
+	spiffeidv1beta1 "github.com/spiffe/spire/api/spiffecrd/v1beta1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// PodReconcilerConfig holds the config passed in when creating the reconciler
+type PodReconcilerConfig struct {
+	Cluster            string
+	Ctx                context.Context
+	DisabledNamespaces []string
+	Log                logrus.FieldLogger
+	Mgr                ctrl.Manager
+	PodLabel           string
+	PodAnnotation      string
+	TrustDomain        string
+}
+
+// PodReconciler holds the runtime configuration and state of this controller
+type PodReconciler struct {
+	client.Client
+	c      PodReconcilerConfig
+	scheme *runtime.Scheme
+}
+
+// NewPodReconciler creates a new PodReconciler object
+func NewPodReconciler(config PodReconcilerConfig) (*PodReconciler, error) {
+	r := &PodReconciler{
+		Client: config.Mgr.GetClient(),
+		c:      config,
+		scheme: config.Mgr.GetScheme(),
+	}
+
+	err := ctrl.NewControllerManagedBy(config.Mgr).
+		For(&corev1.Pod{}).
+		Complete(r)
+	if err != nil {
+		return nil, err
+	}
+
+	return r, nil
+}
+
+// Reconcile creates a new SPIFFE ID when pods are created
+func (r *PodReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	if containsString(r.c.DisabledNamespaces, req.NamespacedName.Namespace) {
+		return ctrl.Result{}, nil
+	}
+
+	pod := corev1.Pod{}
+	ctx := r.c.Ctx
+
+	if err := r.Get(ctx, req.NamespacedName, &pod); err != nil {
+		if !errors.IsNotFound(err) {
+			r.c.Log.WithError(err).Error("Unable to get Pod")
+			return ctrl.Result{}, err
+		}
+
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// Pod neeeds to be assigned a node before it can get a SPIFFE ID
+	if pod.Spec.NodeName == "" {
+		return ctrl.Result{}, nil
+	}
+
+	_, err := r.createPodEntry(ctx, &pod)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// getOrCreatePodEntry attempts to create a new SpiffeID resource. if the entry already exists, it returns it.
+func (r *PodReconciler) createPodEntry(ctx context.Context, pod *corev1.Pod) (*spiffeidv1beta1.SpiffeID, error) {
```

Comment:
The returned SpiffeID is not used. How would you feel about not returning it?

Reply:
Makes sense, updated: https://github.com/spiffe/spire/pull/1581/commits/b60690d2b925bf1cfb8cb45efd6a4ea6bd9e2ff3